submissions/5mountains/dom-api-task/js/main.js

```diff
@@ -0,0 +1,102 @@
+const styles = {
+    menuLink: 'menu__link',
+    menuLinkActive: 'menu__link--active',
+    menuItem: 'menu__item',
+    heroItem: 'hero__item',
+    heroItemActive: 'hero__item--active',
+    heroImg: 'hero__image',
+    itemIcon: 'item__icon',
+    itemTitle: 'item__caption',
+    itemPrice: 'item__price',
+    itemInfo: 'item__description'
+};
+const sectionMenu = document.querySelector('.nav__menu');
+const sectionHero = document.querySelector('.hero');
+const url = './data/data.json';
+let selectMenuItem;
+let selectHeroItem;
+
+async function fetchAsync () {
+    const response = await fetch(url);
+    const data = await response.json();
+    return data;
+}   
+  
+fetchAsync()
+    .then(data => {
+        const details = createHtml(data);
+        drawPage(details);
+        toggleTabs();
+    })
+    .catch(reason => console.log(reason.message));
+
+function createHtml(data) {
+    let navContent  = '';
+    let heroContent  = '';
+    data.pancakes.forEach((elem, index) => {
+        const menuItems = generateNavContent(elem, index); 
+        const heroItems = generateHeroContent(elem, index); 
+        navContent += menuItems;
+        heroContent += heroItems;
+    });
+    return {
+        navContent,
+        heroContent
+    };
+}
+
+function generateNavContent(elem, index) {
+    const navLink = `<a class=${styles.menuLink} href='#' id=link${index+1}>${elem.title}</a>`;
```

Comment:
Let's return the template string without creating variables. It will be more readable, to my mind.
Also, don't forget about checking the input parameters of this function for truthy. In this case, it's acceptable, but in the future keep this in mind.
```suggestion
    return `<li class=${styles.menuItem}><a class=${styles.menuLink} href='#' id=link${index+1}>${elem.title}</a></li>`;
```

Reply:
Hmm, okay, in this case, it's a good idea.